### PR TITLE
Debug traces for TA commands and introduce ID/string helpers

### PR DESCRIPTION
**Pull request by etienne-lms**

### CODE REVIEW

**Comment by etienne-lms** [comment]
Issue reported by checkpatch, false positive IMO in this case:
```
CHECK: Macro argument reuse 'table' - possible side-effects?
#64: FILE: ta/pkcs11/src/pkcs11_helpers.c:32:
+#define ID2STR(id, table, prefix)	\
+	id2str(id, table, ARRAY_SIZE(table), prefix)
```

**Comment by etienne-lms** [comment]
pushed wrong ref... updating...

File: ta/pkcs11/src/pkcs11_helpers.c [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+#include <util.h>
+
+#include "pkcs11_helpers.h"
+
+static const char __maybe_unused unknown[] = "<unknown-identifier>";
+
+struct any_id {
+	uint32_t id;
+#if CFG_TEE_TA_LOG_LEVEL > 0
+	const char *string;
```
Comment (jenswi-linaro) [inline ta/pkcs11/src/pkcs11_helpers.c]: What is the point with `id` above if this isn't defined?

File: ta/pkcs11/src/entry.c [hunk 2]
```diff
@@ -136,4 +149,12 @@ TEE_Result TA_InvokeCommandEntryPoint(void *tee_session __unused, uint32_t cmd,
 		ctrl->memref.size = 0;
 
 	return res;
+
+bad_types:
+	DMSG("Bad parameter types used at PKCS11 TA entry:");
```
Comment (jenswi-linaro) [inline ta/pkcs11/src/entry.c]: I'm not sure this "usage" helps very much. The only expected caller is from within the libcktee and that caller is supposed to know this stuff already.

Comment (etienne-lms) [reply ta/pkcs11/src/entry.c]: true. will discard the change.

File: ta/pkcs11/src/pkcs11_helpers.c (hunk 1 above)
Comment (etienne-lms) [reply ta/pkcs11/src/pkcs11_helpers.c]: This structure is used (later) for ID to string conversion and also for valid IDs lists.
In this P-R it is used to trace invoked commands and unused when traces are disabled.
In a later change, the ID will allow to list valid attribute IDs, or valid mechanism IDs. For those, the array will be used both for valid ID test and for (debug) string conversion.

Comment (jenswi-linaro) [reply ta/pkcs11/src/pkcs11_helpers.c]: Fair enough

**Comment by jenswi-linaro** [comment]
With the "usage" part removed:
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by etienne-lms** [comment]
command address.
build failure (on log level) reported by shippable fixed.
R-b tag applied.

**Pull request merged**